### PR TITLE
Raise error on bad request (non 200 resp)

### PR DESCRIPTION
**Pull request by Loberation**
Raise Exception if response type for request is not okay. Pass body of HTTP:Response to the Error.

#9

### CODE REVIEW

**Comment by Loberation** [comment]
@rnestler

File: spec/system/simple_query_spec.rb [hunk 1]
```diff
@@ -44,6 +44,18 @@
...
+      let(:what) { "071 2222 777" }
+
+      it "raises an exception" do
+        expect { subject }.to raise_error(TelSearch::Error)
+      end
+
+      it "reports the error body in the TelSearch::Error body" do
+        expect { subject }.to raise_error (/API-Key is invalid/)
+      end
+    end
```
Comment (rnestler) [inline spec/system/simple_query_spec.rb:57]: I'd just use one VCR cassette, since we test the same thing (I noticed that we could do this in other specs as well, but this is outside of the scope of this PR).

Also we should set `API_KEY` explicitly to an invalid value, so re-recording of the VCR works:
```
let(:client) { TelSearch::Client.new('invalid-key') }
```

**Review by rnestler — APPROVED** [review]
Thanks!
